dashboard/test/widgets/task_grid_test.dart

```diff
@@ -95,6 +96,77 @@ void main() {
     buildState.dispose();
   });
 
+  testWidgets('TaskGridContainer supports mouse drag', (WidgetTester tester) async {
+    await precacheTaskIcons(tester);
+    final DevelopmentCocoonService service = DevelopmentCocoonService(DateTime.utc(2020));
+    final BuildState buildState = BuildState(
+      cocoonService: service,
+      authService: MockGoogleSignInService(),
+    );
+    void listener1() {}
+    buildState.addListener(listener1);
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: ValueProvider<BuildState>(
+          value: buildState,
+          child: const Material(
+            child: TaskGridContainer(),
+          ),
+        ),
+      ),
+    );
+    await tester.pump();
+
+    final int commitCount = tester.elementList(find.byType(CommitBox)).length;
+    expect(commitCount, 16); // based on screen size this is how many show up
+
+    final double xPosition = tester.getTopLeft(find.byType(CommitBox).first).dx;
+
+    for (int index = 0; index < commitCount; index += 1) {
+      // All the x positions should match the first instance if they're all in the same column
+      expect(tester.getTopLeft(find.byType(CommitBox).at(index)).dx, xPosition);
+    }
+
+    await expectGoldenMatches(find.byType(TaskGrid), 'task_grid_test.dev.origin.png');
+
+    // Check if the LOADING... indicator appears.
+    service.paused = true;
+
+    TestGesture gesture = await tester.startGesture(
+      tester.getCenter(find.byType(TaskGrid)),
+      kind: PointerDeviceKind.mouse,
+    );
+    await gesture.moveBy(const Offset(0.0, -5000.0));
+    await gesture.up();
+
+    await tester.pumpAndSettle();
+    await expectGoldenMatches(find.byType(TaskGrid), 'task_grid_test.dev.mouse_scroll_y.png');
+
+    await gesture.removePointer();
+    service.paused = false;
+    await tester.pumpAndSettle();
+
+    // Check the right edge after the data comes in.
+    service.paused = true;
+
+    gesture = await tester.startGesture(
+      tester.getCenter(find.byType(TaskGrid)),
+      kind: PointerDeviceKind.mouse,
+    );
+    await gesture.moveBy(const Offset(-5000, 0));
+    await gesture.up();
+
+    await tester.pumpAndSettle();
+    await expectGoldenMatches(find.byType(TaskGrid), 'task_grid_test.dev.mouse_scroll_x.png');
```

Comment:
Is this moving the lattice? The goldens seem to be the origin location without any scroll